File: springboot/ps5GameTradingSystem/src/main/java/com/zxy/ps5gametradingsystem/entity/Order.java
package com.zxy.ps5gametradingsystem.entity;

import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.util.List;

@Data
@TableName("`order`")
public class Order {
    private String id;
    private String userId;
    private String orderTime;
    private String orderAddress;
    private Integer sum;
    // 该字段在数据库表中不存在，用于接收关联查询的结果
    @TableField(exist = false)
    private List<Order_details> orderDetails;
}

File: vue/ps5GameTradingSystem/src/types/order.ts
export interface order {
    id:   string,
    userId: string,
    orderTime:  string,
    orderAddress: string,
    sum: number,
    orderDetails:orderDetails,
}
export interface orderDetail {
    id:   string,
    orderId: string,
    gameName:  string,
    price: number,
    quantity: number,
}
// 自定义类型
export type orderDetails = Array<orderDetail>
export type orders = order[]